plat/drivers/include/gic/gic-v3.h

```diff
@@ -374,4 +374,12 @@
  */
 int gicv3_probe(const void *fdt, struct _gic_dev **dev);
 
+/**
+ * Send a software generated interrupt to the specified core.
+ * 
+ * @sigintid the software generated interrupt id
+ * @cpuid the id of the targeted cpu
+ */
+void gic_sgi_gen(uint8_t sigintid, uint32_t cpuid);
```

Comment:
I think the best approach for GIC driver is actually add gic_sgi_gen function to struct _gic_operations, then set corresponding static function for both GICv2 and GICv3.

Reply:
I've updated the operations interface.